specifyweb/middleware/general.py

```diff
@@ -51,3 +59,39 @@ def process_exception(self, request, exception) -> Optional[http.HttpResponse]:
             except TypeError as e:
                 exception = SpecifyExceptionWrapper(e)
                 return http.HttpResponse(exception.to_json(), status=exception.status_code)
+
+
+def serialize_django_obj(django_obj: models.QuerySet or models.Model) -> List[Dict[str, Any]] or Dict[str, Any]:
```

Comment:
There is a function in `./specify/api.py` called `_obj_to_data`. Can you check if it is related to what you are doing here? If so, would it make sense to merge that function with this one?

Reply:
Yes I can merge the two. Currently `_obj_to_data` does not handle serializing QuerySets, but I can always repurpose the logic. 

The main concern I have with this is that `_obj_to_data` returns every field of the associated object. If the object has any relationships, it includes every field in the associated table (`_obj_to_data` does the heavy work for creating the responses for the main specify endpoint)

For example, here is a (nonformatted json) agent with 2 addresses:

![serialized_agent](https://user-images.githubusercontent.com/64045831/216140169-c3b628ec-95e5-4ab0-bc26-a760973ef6b8.png)

There is the argument that the data is important, but I feel that (more often than not) most of the data from the function does not need to be viewed and it _does_ clutter a _lot_ of the space and can hinder viewing _all_ of the records. The most important information will be 'lost' in the clutter

Reply:
The question is should we truncate the data to the most 'important' parts and make the data more readable, or show _all_ of the related data but make everything harder to find/identify?